pcdsdevices/inout.py

```diff
@@ -0,0 +1,90 @@
+import math
+
+from .state import StatePositioner, StateRecordPositioner, PVStatePositioner
+
+
+class InOutPositioner(StatePositioner):
+    """
+    Basic InOut StatePositioner. It can be inserted and removed and queried for
+    insertion and removal. It can also define transmission values for the
+    various states.
+
+    Attributes
+    ----------
+    in_states: list of strings
+        State values that should be considered 'in'.
+
+    out_states: list of strings
+        State values that should be considered 'out'.
+
+    _tranmsission: dict{str: number}
+        Mapping from each state to the transition coefficient. This should be a
+        number from 0 to 1. Default values will be 1 (full transmission) for
+        out_states, 0 (full block) for in_states, and nan (no idea!) for
+        unaccounted states.
+    """
+    states_list = ['IN', 'OUT']
+    in_states = ['IN']
+    out_states = ['OUT']
+    _transmission = {}
+
+    def __init__(self, prefix, *, name, **kwargs):
+        super().__init__(prefix, name=name, **kwargs)
+        self._trans_enum = {}
+        self._extend_trans_enum(self.in_states, 0)
+        self._extend_trans_enum(self.out_states, 1)
+
+    @property
+    def inserted(self):
+        return self._pos_in_list(self.in_states)
+
+    @property
+    def removed(self):
+        return self._pos_in_list(self.out_states)
+
+    def insert(self, moved_cb=None, timeout=None, wait=False):
+        """
+        Macro to move this device to the first state on the in_states list.
+        """
+        return self.move(self.in_states[0], moved_cb=moved_cb,
+                         timeout=timeout, wait=wait)
+
+    def remove(self, moved_cb=None, timeout=None, wait=False):
+        """
+        Macro to move this device to the first state on the out_states list.
+        """
+        return self.move(self.out_states[0], moved_cb=moved_cb,
+                         timeout=timeout, wait=wait)
+
+    @property
+    def transmission(self):
+        state = self.get_state(self.position)
+        return self._trans_enum.get(state, math.nan)
+
+    def _extend_trans_enum(self, state_list, default):
+        for state in state_list:
+            enumst = self.get_state(state)
+            self._trans_enum[enumst] = self._transmission.get(state, default)
+
+    def _pos_in_list(self, state_list):
+        current_state = self.get_state(self.position)
+        for state in state_list:
+            if current_state == self.get_state(state):
+                return True
+        return False
+
+
+class InOutRecordPositioner(StateRecordPositioner, InOutPositioner):
+    pass
```

Comment:
Might be nice to have a little docstring here at least. I realize that it will probably be redundant with the `StateRecordPositioner`, `StatePositioner`. but I'm actually guessing that more developer will interface with this class that the plain positioner class. I vote either right something short or reference a shared explanation